commit 7c222c89366e49b10a337a4a4d4a957f27ec6b8d
Author: Jamie Coldhill <kodi@Ubuntu.12.04.5>
Date:   Sat Jan 16 23:05:25 2016 +0800

    Fix.AML.SysInfo.temp.patch

diff --git a/xbmc/utils/CPUInfo.cpp b/xbmc/utils/CPUInfo.cpp
index 660d355..833940c 100644
--- a/xbmc/utils/CPUInfo.cpp
+++ b/xbmc/utils/CPUInfo.cpp
@@ -99,6 +99,12 @@
 
 #include "utils/StringUtils.h"
 
+#ifdef HAS_LIBAMCODEC
+#include "utils/AMLUtils.h"
+#endif
+
+ using namespace std;
+
 // In milliseconds
 #define MINIMUM_TIME_BETWEEN_READS 500
 
