From 792e657849284feb01e856482d450912ad40e79f Mon Sep 17 00:00:00 2001
From: alex <surkov.alex@gmail.com>
Date: Thu, 25 Feb 2016 08:36:09 +0100
Subject: [PATCH] Fix logo after TV off/on

---
 drivers/amlogic/display/logo/logo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/display/logo/logo.c b/drivers/amlogic/display/logo/logo.c
index 534c2c4..62a5632 100644
--- a/drivers/amlogic/display/logo/logo.c
+++ b/drivers/amlogic/display/logo/logo.c
@@ -213,7 +213,7 @@ static int refresh_mode_and_logo(bool first)
 		}
 		last_mode = cur_mode;
 		vout_notifier_call_chain(VOUT_EVENT_MODE_CHANGE, &cur_mode);
-		set_osd_freescaler(logo_info.index, cur_mode);
+		//set_osd_freescaler(logo_info.index, cur_mode);//vit commented
 	}
 
 	return 0;
