From 1bfcb716d0e6654bc7c7ad3c723166675c6cdb59 Mon Sep 17 00:00:00 2001
From: Radostan Riedel <raybuntu@googlemail.com>
Date: Sun, 5 Jun 2016 10:17:04 +0200
Subject: [PATCH] [aocec] Add option to disable cec completely

---
 drivers/amlogic/cec/hdmi_ao_cec.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/cec/hdmi_ao_cec.c b/drivers/amlogic/cec/hdmi_ao_cec.c
index 21ca8bb..a21faf7 100644
--- a/drivers/amlogic/cec/hdmi_ao_cec.c
+++ b/drivers/amlogic/cec/hdmi_ao_cec.c
@@ -996,6 +996,9 @@ int cec_node_init(struct hdmitx_dev *hdmitx_device)
     if ((cec_dev->hal_flag & (1 << HDMI_OPTION_SYSTEM_CEC_CONTROL)))
         return -1;
 
+    if (!(hdmitx_device->cec_func_config & (1 << CEC_FUNC_MSAK)))
+        return -1;
+
     CEC_INFO("cec_node_init started\n");
 
     cec_phy_addr = ((a << 12) | (b << 8) | (c << 4) | (d << 0));
