diff --git a/xbmc/view/ViewStateSettings.cpp b/xbmc/view/ViewStateSettings.cpp
index cf3beb3..aa83b0e 100644
--- a/xbmc/view/ViewStateSettings.cpp
+++ b/xbmc/view/ViewStateSettings.cpp
@@ -124,12 +124,14 @@ bool CViewStateSettings::Load(const TiXmlNode *settings)
   pElement = settings->FirstChild(XML_GENERAL);
   if (pElement != NULL)
   {
+    /*
+    // Force setting level to expert
     int settingLevel;
     if (XMLUtils::GetInt(pElement, XML_SETTINGLEVEL, settingLevel, (const int)SettingLevelBasic, (const int)SettingLevelExpert))
       m_settingLevel = (SettingLevel)settingLevel;
     else
-      m_settingLevel = SettingLevelStandard;
-
+    */
+      m_settingLevel = SettingLevelExpert;
     const TiXmlNode* pEventLogNode = pElement->FirstChild(XML_EVENTLOG);
     if (pEventLogNode != NULL)
     {
