From 394f640f59be50155aec2afc01911ce88d38be0a Mon Sep 17 00:00:00 2001
From: "Chris \"koying\" Browet" <cbro@semperpax.com>
Date: Thu, 12 May 2016 18:28:56 +0200
Subject: [PATCH] FIX: [webserver] extend balcklist

---
 xbmc/utils/FileUtils.cpp | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/xbmc/utils/FileUtils.cpp b/xbmc/utils/FileUtils.cpp
index e13db0e..523b594 100644
--- a/xbmc/utils/FileUtils.cpp
+++ b/xbmc/utils/FileUtils.cpp
@@ -171,20 +171,20 @@ bool CFileUtils::ZebraListAccessCheck(const std::string &filePath)
 {
   // white/black list access checks, disallow exploits
 
-  // no access to the passwords.xml file,
-  // this can expose user/pass of remote servers
-  if (filePath.find("passwords.xml") != std::string::npos)
+  static const char *blacklist_access[] = {
+    "passwords.xml",
+    "sources.xml",
+    "guisettings.xml",
+    "advancedsettings.xml",
+    NULL
+  };
+  for (const char **ptr = blacklist_access; *ptr; ptr++)
   {
-    CLog::Log(LOGDEBUG,"http access denied");
-    return false;
-  }
-
-  // no access to the sources.xml file,
-  // this can expose user/pass of remote servers
-  if (filePath.find("sources.xml") != std::string::npos)
-  {
-    CLog::Log(LOGDEBUG,"http access denied");
-    return false;
+    if (filePath.find(*ptr) != std::string::npos)
+    {
+      CLog::Log(LOGDEBUG,"http access denied");
+      return false;
+    }
   }
 
 #if defined(TARGET_DARWIN)
