commit 68e0d985f0460df8d63db4806f4f6acb6db2c007
Author: Jamie Coldhill <wrxtasy@amnet.net.au>
Date:   Sun May 1 17:33:29 2016 +0800

    NFSFile: Change chunksize to 1MB

diff --git a/xbmc/filesystem/NFSFile.h b/xbmc/filesystem/NFSFile.h
index fc474e5..9be3493 100644
--- a/xbmc/filesystem/NFSFile.h
+++ b/xbmc/filesystem/NFSFile.h
@@ -147,7 +147,7 @@ namespace XFILE
     //implement iocontrol for seek_possible for preventing the stat in File class for
     //getting this info ...
     virtual int IoControl(EIoControl request, void* param){ if(request == IOCTRL_SEEK_POSSIBLE) return 1;return -1;};    
-    virtual int  GetChunkSize() {return 1;}
+    virtual int  GetChunkSize() {return 1024*1024;}
     
     virtual bool OpenForWrite(const CURL& url, bool bOverWrite = false);
     virtual bool Delete(const CURL& url);
